src/services/creategeojsonbloodhound.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @fileoverview Provides a function that creates a Bloodhound engine
+ * expecting GeoJSON responses from the search web service, and creating
+ * `ol.Feature` objects as suggestions.
+ *
+ * Example:
+ *
+ * var bloodhound = ngeoCreateGeoJSONBloodhound(
+ *   'http://example.com/fulltextsearch?query=%QUERY',
+ *   ol.proj.get('EPSG:3857'));
+ * bloodhound.initialize();
+ */
+
+goog.provide('ngeo.CreateGeoJSONBloodhound');
+
+goog.require('ngeo');
+goog.require('ol.format.GeoJSON');
+
+
+/**
+ * @typedef {function(string, ol.proj.Projection=,
+ *     ol.proj.Projection=):Bloodhound}
+ */
+ngeo.CreateGeoJSONBloodhound;
+
+
+/**
+ * @param {string} url URL pattern.
+ * @param {ol.proj.Projection=} opt_featureProjection Feature projection.
+ * @param {ol.proj.Projection=} opt_dataProjection Data projection.
+ * @return {Bloodhound} The Bloodhound object.
+ */
+ngeo.createGeoJSONBloodhound = function(url, opt_featureProjection,
+    opt_dataProjection) {
+  var geojsonFormat = new ol.format.GeoJSON();
+  return new Bloodhound(/** @type {BloodhoundOptions} */ ({
+    remote: {
+      url: url,
+      ajax: {
+        dataType: 'jsonp'
+      },
```

Comment:
> We are setting the limit option and using a replace function to handle localization.

Yes, I see.

And your `filter` function doesn't create `ol.Feature` objects either. So this Bloodhound factory is not useful to you…

This factory is really about creating a `Bloodhound` engine with a `filter` function returning `ol.Feature` created out of GeoJSON documents. The rest should be customizable through options.

What about the following?

```js
/**
 * @params {BloodhoundOptions|string} options Bloodhound options or a URL. If a string is passed the
 *     default options are used, with this string as the `url` in the `remote` object.
 * @param {ol.proj.Projection=} opt_featureProjection Feature projection.
 * @param {ol.proj.Projection=} opt_dataProjection Data projection.
 * @return {Bloodhound} The Bloodhound object.
 */
ngeo.createGeoJSONBloodhound = function(
    options, opt_featureProjection, opt_dataProjection) {
  var bloodhoundOptions = /** @type BloodhoundOptions */ ({
    remote: {
      url: goog.isString(options) ? options : '',
      ajax: {
        dataType: 'jsonp'
      },
      filter: function(parsedResponse) {
        // …
      }
    },
    datumTokenizer: goog.nullFunction,
    queryTokenizer: Bloodhound.tokenizers.whitespace
  };
  if (!goog.isString(options)) {
    goog.object.extend(bloodhoundOptions, options);
  }
  return new Bloodhound(bloodhoundOptions);
};
```

I don't know if it's worth it. Maybe the user should just create his Bloodhound engine himself if he needs something more specific than what this factory does.

Thoughts?


> Would it be possible to update the remote url of a Bloodhound afterwards?

Not that I know.